File: Day-34/136A-Presents.cpp
#include <iostream>
using namespace std;

int main() {
    int n;
    cin >> n;
    int p[n];
    for (int i = 0; i < n; i++) {
        cin >> p[i];
    }
    int ans[n];
    for (int i = 0; i < n; i++) {
        int recipient = p[i] - 1; // 0-indexed index of the recipient
        ans[recipient] = i + 1;
    }
    for (int i = 0; i < n; i++) {
        cout << ans[i] << " ";
    }
    cout << endl;
    return 0;
}

File: Day-34/136A-Presents.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] p = new int[n];
        for (int i = 0; i < n; i++) {
            p[i] = scanner.nextInt();
        }
        int[] ans = new int[n];
        for (int i = 0; i < n; i++) {
            int recipient = p[i] - 1;
            ans[recipient] = i + 1;
        }
        for (int i = 0; i < n; i++) {
            System.out.print(ans[i] + " ");
        }
        System.out.println();
    }
}
